Tests/CommandsTests/WorkspaceTests.swift

```diff
@@ -374,12 +372,138 @@ final class WorkspaceTests: XCTestCase {
+            XCTAssertFalse(exists(workspace.editablesPath))
+        }
+    }
+
+    func testUneditDependency() throws {
+        mktmpdir { path in
+            let manifestGraph = try MockManifestGraph(at: path,
+                rootDeps: [
+                    MockDependency("A", version: Version(1, 0, 0)..<Version(1, .max, .max)),
+                ],
+                packages: [
+                    MockPackage("A", version: v1),
+                    MockPackage("A", version: nil), // To load the edited package manifest.
+                ]
+            )
+            // Create the workspace.
+            let workspace = try Workspace(rootPackage: path, manifestLoader: manifestGraph.manifestLoader, delegate: TestWorkspaceDelegate())
+            // Load the package graph.
+            let graph = try workspace.loadPackageGraph()
+            // Sanity checks.
+            XCTAssertEqual(graph.packages.count, 2)
+            XCTAssertEqual(graph.packages.map{ $0.name }.sorted(), ["A", "Root"])
+
+            let manifests = try workspace.loadDependencyManifests()
+            guard let aManifest = manifests.lookup("A") else {
+                return XCTFail("Expected manifest for package A not found")
+            }
+            // Put the dependency in edit mode.
+            let dependency = try workspace.findDependency(forURL: aManifest.url)!
+            try workspace.edit(dependency: dependency, at: dependency.currentRevision, packageName: aManifest.name)
+
+            let editedDependency = try workspace.findDependency(forURL: aManifest.url)!
+            let editRepoPath = workspace.editablesPath.appending(editedDependency.subpath)
+            // Write something in repo.
+            try localFileSystem.writeFileContents(editRepoPath.appending(component: "test.txt"), bytes: "Hi")
+            try systemQuietly([Git.tool, "-C", editRepoPath.asString, "add", "test.txt"])
+
+            // Try to unedit.
+            do {
+                try workspace.unedit(dependency: editedDependency, forceRemove: false)
+                XCTFail("Unexpected edit success")
+            } catch WorkspaceOperationError.hasUncommitedChanges(let repo) {
+                XCTAssertEqual(repo, editRepoPath)
+            }
+            // Commit and try to unedit.
+            // FIXME: Create proper utility for this.
+            try systemQuietly([Git.tool, "-C", editRepoPath.asString, "config", "user.email", "example@example.com"])
+            try systemQuietly([Git.tool, "-C", editRepoPath.asString, "config", "user.name", "Example Example"])
+            try systemQuietly([Git.tool, "-C", editRepoPath.asString, "commit", "-m", "Add some files."])
+            do {
+                try workspace.unedit(dependency: editedDependency, forceRemove: false)
+                XCTFail("Unexpected edit success")
+            } catch WorkspaceOperationError.hasUnpushedChanges(let repo) {
+                XCTAssertEqual(repo, editRepoPath)
+            }
```

Comment:
There is a test missing to push changes and try to remove without force which is TODO but otherwise should be ok to review